src/hal/src/pso/mod.rs

```diff
@@ -221,32 +248,112 @@ pub struct SpecializationConstant {
 }
 
 /// Specialization information structure.
-#[derive(Debug, Copy)]
-pub struct Specialization<'a> {
+#[derive(Debug, Copy, Clone)]
+pub struct RawSpecialization<'a> {
     /// Constant array.
     pub constants: &'a [SpecializationConstant],
     /// Raw data.
     pub data: &'a [u8],
 }
 
-impl<'a> Default for Specialization<'a> {
+impl RawSpecialization<'_> {
+    /// Create empty specialization instance.
+    pub fn new() -> Self {
```

Comment:
do we still need this?

Reply:
I don't really like types without `new` without reason like impossibility to implement.